common/quotas/ratelimiter.go

```diff
@@ -28,13 +28,19 @@ import (
 	"time"
 
 	"golang.org/x/time/rate"
+
+	"github.com/uber/cadence/common/clock"
 )
 
 const (
 	_defaultRPSTTL = 60 * time.Second
 	_burstSize     = 1
 )
 
+// TODO: this is likely major overkill, especially now that clock.Ratelimiter exists.
```

Comment:
The sad part of this benchmark is that it does not exist. I'm unsure what was tested here and whether it is true or false.

Reply:
yea - the benchmark exists in the commit that introduced this file, but AFAICT it's comparing a _fully_ custom ratelimiter ("tokenbucket.go" file, does not currently exist) against this wrapper around `*rate.Limiter`.

Which, yeah, I'd expect it to be faster.  `*rate.Limiter` is pretty well built for performance purposes.